codex-rs/tui/src/app.rs

```diff
@@ -357,3 +429,119 @@ impl<'a> App<'a> {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    #![allow(clippy::unwrap_used)]
+    use super::*;
+    use codex_core::protocol::AgentMessageEvent;
+    use codex_core::protocol::EventMsg;
+    use codex_core::protocol::TaskCompleteEvent;
+    use pretty_assertions::assert_eq;
+
+    #[test]
+    fn test_create_compact_summary_prompt_with_content() {
+        let summary_text = "User asked about Rust. I explained ownership and borrowing.";
+        let result = create_compact_summary_prompt(summary_text);
+
+        let expected = COMPACT_SUMMARY_TEMPLATE.replace(
```

Comment:
I would just `assert_eq!()` using a string literal that, yes, is a copy of `COMPACT_SUMMARY_TEMPLATE`. Again, I would use `r#`.